hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -2462,7 +2463,7 @@ public <T> T find(Class<T> entityClass, Object primaryKey, LockModeType lockMode
 			This may happen if the entity has an associations mapped with @NotFound(action = NotFoundAction.EXCEPTION)
 			and this associated entity is not found.
 			 */
-			if ( enfe instanceof FetchNotFoundException ) {
+			if ( enfe instanceof FetchNotFoundException || enfe instanceof EntityFilterException ) {
 				throw enfe;
 			}
```

Comment:
orthogonal, but we should use a separate `catch` block here instead of `if instanceof`